jib-plugins-common/src/main/java/com/google/cloud/tools/jib/plugins/common/PluginConfigurationProcessor.java

```diff
@@ -611,6 +627,38 @@ static String getDefaultBaseImage(ProjectProperties projectProperties)
     throw new IncompatibleBaseImageJavaVersionException(11, javaVersion);
   }
 
+  /**
+   * Parses the list of platforms to a set of {@link Platform}.
+   *
+   * @param rawConfiguration raw configuration data
+   * @return the set of parsed platforms.
+   * @throws InvalidPlatformConfigurationException if there exists a plaformConfiguration with a
+   *     null architectureName or osName
+   */
+  @VisibleForTesting
+  static Set<Platform> getPlatformsSet(RawConfiguration rawConfiguration)
+      throws InvalidPlatformConfigurationException {
+    Set<Platform> platforms = new LinkedHashSet<>();
+    for (PlatformConfiguration platformConfiguration : rawConfiguration.getPlatforms()) {
+      try {
+        Preconditions.checkNotNull(
+            platformConfiguration.getArchitectureName(), "platforms set cannot be empty");
+        Preconditions.checkNotNull(
+            platformConfiguration.getOsName(), "platforms set cannot be empty");
+        Platform platform =
+            new Platform(
+                platformConfiguration.getArchitectureName().orElse(null),
+                platformConfiguration.getOsName().orElse(null));
+        platforms.add(platform);
+      } catch (IllegalArgumentException exception) {
+        throw new InvalidPlatformConfigurationException(
+            platformConfiguration.toString(), platformConfiguration.toString(), exception);
```

Comment:
`platformConfiguration.toString()` will returns some weird string value (for example, `PlatformConfiguration@6d06d69c`), unless the class (or any of its super-classes) overrode and implemented `toString()`.

And I think we don't want to implement `toString()` in `PlatformConfiguration` implementations. So probably we should assemble a human-readable string form here. For example,

`"architecture=" + platformConfiguration.getArchitectureName().orElse("<missing>") + ", os=" + platformConfiguration.getOsName().orElse("<missing>")`

Reply:
I see what you mean .I agree with you on the string implementation. Done